kubernetes-kit-starter/src/test/java/com/vaadin/kubernetes/starter/sessiontracker/SessionSerializerTest.java

```diff
@@ -200,6 +204,39 @@ void serialize_optimisticLocking_sessionLockRequired_immediatelySwitchToPessimis
         verify(connector).sendSession(notNull());
     }
 
+    @Test
+    void serialize_and_deserialize_unserializableComponentWrapper() {
+        UnserializableComponentWrapper<State, Unserializable> wrapper = createUnserializableComponentWrapper();
+        vaadinSession.lock();
+        UI ui = new MockedUI();
+        ui.getInternals().setSession(vaadinSession);
+        ui.add(wrapper);
+        ui.doInit(null, 1234, "appId");
+        vaadinSession.addUI(ui);
+        vaadinSession.unlock();
+
+        AtomicBoolean serializationCompleted = new AtomicBoolean();
+        doAnswer(i -> serializationCompleted.getAndSet(true)).when(connector)
+                .markSerializationComplete(clusterSID);
+
+        List<SessionInfo> infoList = new ArrayList<>();
+        doAnswer(i -> infoList.add(i.getArgument(0))).when(connector)
+                .sendSession(any());
+
+        vaadinSession.setLockTimestamps(10, 20);
+
+        serializer.serialize(httpSession);
+        await().atMost(1000, MILLISECONDS).untilTrue(serializationCompleted);
+        verify(serializationCallback).onSerializationSuccess();
+
+        try {
+            serializer.deserialize(infoList.get(0), httpSession);
```

Comment:
The test can verify that the unserializable component is present in the deserialized UI and contains the expected data.

Reply:
Updated the test.